packages/frontend/src/components/QrReader/index.tsx

```diff
@@ -129,25 +158,40 @@ export default function QrReader({ onError, onScan }: Props) {
     getAllCameras()
   }, [])
 
-  // General handler for scanning results coming from the "jsqr" library.
-  //
-  // Additionally we have checks in place to make sure we're not firing any
-  // callbacks when this React component has already been unmounted.
-  const handleScanResult = useCallback(
-    result => {
-      let unmounted = false
-
-      if (unmounted) {
-        return
+  const handlePasteFromClipboard = useCallback(async () => {
+    try {
+      const result = await processClipBoard(runtime)
+      if (typeof result === 'string') {
+        onScan(result)
+      } else {
+        try {
+          onScan(result.data)
+        } catch (error) {
+          console.error('Error while processing clipboard content:', error)
+        }
```

Comment:
If an error occurs here, `userFeedback` doesn't get called. I believe this is not the case in the pre-MR version, where if `handleScanResult` throws, we do call `userFeedback`. Is this what we want?

Reply:
onScan has its own error handling anyway so this is not needed

Reply:
I pass the error now always to onError so the parent component shows an AlertDialog